src/components/processor-factory/form-processor-factory.component.tsx

```diff
@@ -0,0 +1,108 @@
+import { type FormProcessorContextProps } from '../../types';
+import { useFormFieldsMeta } from '../../hooks/useFormFieldsMeta';
+import { useFormFactory } from '../../provider/form-factory-provider';
+import { useFormFieldValueAdapters } from '../../hooks/useFormFieldValueAdapters';
+import { EncounterFormProcessor } from '../../processors/encounter/encounter-form-processor';
+
+interface FormProcessorFactoryProps {
+  formJson: FormSchema;
+  isSubForm: boolean;
+}
+
+const FormProcessorFactory = ({ formJson, isSubForm = false }: FormProcessorFactoryProps) => {
+  const { patient, sessionMode, formProcessors, layoutType, location, provider, sessionDate, visit } = useFormFactory();
+
+  // TODO: load processor from the registry
+  const processor = useMemo(() => {
+    const ProcessorClass = formProcessors[formJson.processor];
+    if (processor) {
+      return processor;
+    }
+    if (ProcessorClass) {
+      return new ProcessorClass(formJson);
+    }
+    console.error(`Form processor ${formJson.processor} not found, defaulting to EncounterFormProcessor`);
+    return new EncounterFormProcessor(formJson);
+  }, [formProcessors, formJson.processor]);
+
+  const [processorContext, setProcessorContext] = useState<FormProcessorContextProps>({
+    patient,
+    formJson,
+    sessionMode,
+    layoutType,
+    location,
+    currentProvider: provider,
+    processor,
+    sessionDate,
+    visit,
+  });
+  const { formFields: rawFormFields, conceptReferences } = useFormFields(formJson);
+  const { concepts: formFieldsConcepts, isLoading: isLoadingConcepts } = useConcepts(conceptReferences);
+  const formFieldsWithMeta = useFormFieldsMeta(rawFormFields, formFieldsConcepts);
+  const formFieldAdapters = useFormFieldValueAdapters(rawFormFields);
+  const formFieldValidators = useFormFieldValidators(rawFormFields);
+  const { isLoading: isLoadingCustomDeps } = useProcessorDependencies(processor, processorContext, setProcessorContext);
+
+  const useCustomHooks = processor.getCustomHooks().useCustomHooks;
+  const [isLoadingCustomHooks, setIsLoadingCustomHooks] = useState(!!useCustomHooks);
+
+  const {
+    isLoadingInitialValues,
+    initialValues,
+    error: initialValuesError,
+  } = useInitialValues(processor, isLoadingCustomDeps || isLoadingCustomHooks, processorContext);
+
+  // TODO: handle states individually?
+  useEffect(() => {
+    if (formFieldAdapters) {
+      setProcessorContext((prev) => ({
+        ...prev,
+        ...{ formFieldAdapters },
+      }));
+    }
+    if (formFieldValidators) {
+      setProcessorContext((prev) => ({
+        ...prev,
+        ...{ formFieldValidators },
+      }));
+    }
+    if (formFieldsWithMeta?.length) {
+      setProcessorContext((prev) => ({
+        ...prev,
+        ...{ formFields: formFieldsWithMeta },
+      }));
+    } else if (rawFormFields?.length) {
+      setProcessorContext((prev) => ({
+        ...prev,
+        ...{ formFields: rawFormFields },
+      }));
+    }
```

Comment:
Maybe something like this:

```suggestion
      setProcessorContext((prev) => merge({}, prev, formFieldAdapters ?? {}, formFieldValidators ? {}, formFieldsWithMeta?.length ? {formFields: formFieldsWithMeta} : {}, rawFormFields?.length ? { formFields: rawFormFields } : {}));
```

Reply:
`merge` from `lodash` to support deep merging, but otherwise `Object.assign()` would be the same as this with many fewer `setState()` calls.

Reply:
This is a good solution, but it ends up losing my object references in memory due to the deep clone. This is problematic because I would have to manually reconcile the fields in the formJson tree versus the flattened fields. This can lead to increased memory usage and potential inconsistencies, as I will end up with two separate instances for each form field.

I've ended doing something like:

```ts
setProcessorContext((prev) => ({
      ...prev,
      ...(formFieldAdapters && { formFieldAdapters }),
      ...(formFieldValidators && { formFieldValidators }),
      ...(formFieldsWithMeta?.length
        ? { formFields: formFieldsWithMeta }
        : rawFormFields?.length
        ? { formFields: rawFormFields }
        : {}),
    }));

```

Reply:
Ah! Fair enough. Still... involves fewer calls to the `setProcessContext()`, which is basically the whole point.